nf_core/lint.py

```diff
@@ -335,14 +391,62 @@ def check_version_consistency(self):
         self.passed.append((7, "Version tags are numeric and consistent between container, release tag and config."))
 
 
+    def check_conda_env_yaml(self):
+        """ Check that the conda environment file is valid
+
+        Make sure that a name is given and is consistent with the pipeline name
+        Check that depedency versions are pinned
+        Warn if dependency versions are not the latest available """
+
+        if 'environment.yml' not in self.files:
+            return
+
+        # Check that the environment name matches the pipeline name
+        if self.conda_config['name'] != 'nfcore-{}'.format(self.pipeline_name):
+            self.failed.append((8, "Conda environment name is not consistent ('{}', should be 'nfcore-{}')".format(self.conda_config['name'], self.pipeline_name)))
+        else:
+            self.passed.append((8, "Conda environment name was consistent ('{}')".format(self.conda_config['name'])))
+
+        # Check conda dependency list
+        for dep in self.conda_config.get('dependencies', []):
+            if isinstance(dep, str):
```

Comment:
you now exlude sublists, right? For pip we could use the PyPi JSON API: `http://pypi.python.org/pypi/<package_name>/json` but probably in a separate issue, if you want ;)

Reply:
Sounds like a nice addition! It might be nice to generally try to avoid using `pip` (instead package for conda), but I guess this would be a good bonus addition anyway..

Reply:
I am not sure about the Jaccard index of both package repositories, maybe we can check that first and then decide

Reply:
I am not sure about the Jaccard index of both package repositories, maybe we can check that first and then decide :)

Reply:
referenced this is #29

Reply:
I think take this in a separate PR 👍